### PR TITLE
[SALEOR-3965] Extend Sale & Voucher to include Metadata #7653

### PR DESCRIPTION
**Pull request by piotrgrundas**
I want to merge this change because I want to add support for metadata & private metadata for sales and vouchers

[Saleor PR](https://github.com/mirumee/saleor/pull/7653)

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [x] Attributes `[data-test-id]` are added for new elements
4. [x] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://saleor-3965-sales-voucher-metadata.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3965 Extend Sale & Voucher to include Metadata](https://app.clickup.com/t/2549495/SALEOR-3965)

File: src/discounts/views/SaleDetails/handlers.ts [hunk 1]
```diff
@@ -30,22 +30,26 @@ export function createUpdateHandler(
...
+          type: discountValueTypeEnum(formData.type)
+        }
+      }).then(({ data }) => data?.saleUpdate.errors ?? []),
 
-    updateChannels({
-      variables: getSaleChannelsVariables(id, formData, saleChannelsChoices)
-    });
+      updateChannels({
+        variables: getSaleChannelsVariables(id, formData, saleChannelsChoices)
+      }).then(({ data }) => data?.saleChannelListingUpdate.errors ?? [])
```
Comment (dominik-zeglen) [inline src/discounts/views/SaleDetails/handlers.ts:50]: The thought of having `then` is itching me but I can't really request changes here

**Review by dominik-zeglen — APPROVED** [review]

File: src/discounts/views/SaleDetails/handlers.ts (hunk 1 above)
Comment (piotrgrundas) [reply src/discounts/views/SaleDetails/handlers.ts:50]: True. I could use IIFE with awaits but I think this looks cleaner.
```
      async () => {
        const { data } = await updateChannels({
          variables: getSaleChannelsVariables(id, formData, saleChannelsChoices)
        });
        return data?.saleChannelListingUpdate.errors ?? [];
      }
```

**Review by eCzerniak — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Comment by maarcingebala** [comment]
@jwm0 Related backend PR is merged.

**Pull request merged**